### DISCUSSION TITLE
No hoist options

### DISCUSSION THREAD

**Discussion by billouboq** [pnpm/pnpm#6841] [Q&A]
Hi,

I was trying to migrate from a yarn 1 workspace with one package that needed the "nohoist" options (...react-native package of course), and I could not find how to keep that nohoist and so having all node modules used by that react-native package inside it's own node module. Can we achieve this using pnpm ?

Here is the old package.json options :

```json
"workspaces": {
  "packages": [
	  "packages/*"
  ],
  "nohoist": [
	  "beedeez-app-prepro/**",
	  "beedeez-app-prepro",
  ]
},
```

beedeez-app-prepro being the react-native package

Here is the .npmrc options I tried with pnpm :

```yaml
shamefully-hoist=true
strict-peer-dependencies=false
```

Here is the pnpm-workspace.yml :

```yaml
packages:
  # include packages in subfolders (e.g. apps/ and packages/)
  - 'packages/**'
```

Thank you very much 🙏

**Comment by KSXGitHub** [comment]
Is [public-hoist-pattern](https://pnpm.io/npmrc#public-hoist-pattern) the right option for you?

**Comment by billouboq** [comment]
Feels like it does only work for public package. The react-native package is a private one that use multiple public packages. 

Reading the documentation it says "hoists dependencies matching the pattern to the root modules directory". But a private package in a workspace isn't a dependency by itself so feels like it does not do what's needed.
